tooling/nargo/src/ops/fuzz.rs

```diff
@@ -0,0 +1,349 @@
+use std::path::PathBuf;
+
+use acvm::{
+    acir::native_types::{WitnessMap, WitnessStack},
+    brillig_vm::BranchToFeatureMap,
+    BlackBoxFunctionSolver, FieldElement,
+};
+use noir_greybox_fuzzer::{
+    AcirAndBrilligPrograms, ErrorAndCoverage, FuzzTestResult, FuzzedExecutorExecutionConfiguration,
+    FuzzedExecutorFailureConfiguration, FuzzedExecutorFolderConfiguration, WitnessAndCoverage,
+};
+use noirc_abi::{Abi, InputMap};
+use noirc_driver::{compile_no_check, CompileOptions};
+use noirc_errors::FileDiagnostic;
+use noirc_frontend::hir::{def_map::FuzzingHarness, Context};
+
+use crate::PrintOutput;
+use crate::{
+    errors::try_to_diagnose_runtime_error,
+    foreign_calls::{layers, DefaultForeignCallBuilder},
+    ops::{execute::execute_program_with_brillig_fuzzing, test::TestForeignCallExecutor},
+};
+
+use super::execute_program;
+/// Configuration for fuzzing loop execution
+pub struct FuzzExecutionConfig {
+    /// Number of threads to use for fuzzing
+    pub num_threads: usize,
+    /// Maximum time in seconds to spend fuzzing (default: no timeout)
+    pub timeout: u64,
+}
+
+/// Folder configuration for fuzzing
+pub struct FuzzFolderConfig {
+    /// Corpus folder
+    pub corpus_dir: Option<String>,
+    /// Minimized corpus folder
+    pub minimized_corpus_dir: Option<String>,
+    /// Fuzzing failure folder
+    pub fuzzing_failure_dir: Option<String>,
+}
+
+pub enum FuzzingRunStatus {
+    ExecutionPass,
+    MinimizationPass,
+    CorpusFailure {
+        message: String,
+    },
+    ExecutionFailure {
+        message: String,
+        counterexample: Option<(InputMap, Abi)>,
+        error_diagnostic: Option<FileDiagnostic>,
+    },
+    MinimizationFailure {
+        message: String,
+    },
+    ForeignCallFailure {
+        message: String,
+    },
+    CompileError(FileDiagnostic),
+}
+
+impl FuzzingRunStatus {
+    pub fn failed(&self) -> bool {
+        !matches!(
+            self,
+            FuzzingRunStatus::ExecutionPass | FuzzingRunStatus::MinimizationFailure { .. }
```

Comment:
Is `MinimizationPass` a failure?

If not, it might be good to `match` over self so we can make sure that if a new variant is added, it's handled here.

Reply:
No, this was an error